src/panoramic/cli/husky/service/context.py

```diff
@@ -31,22 +28,21 @@ def dialect(self) -> default.DefaultDialect:
         return RUNTIME_DIALECTS[self._query_runtime]
 
     @classmethod
-    def from_request(cls, data_request: Union[BlendingDataRequest, InternalDataRequest]):
-        if data_request.physical_data_sources:
-            if len(data_request.physical_data_sources) == 1:
-                request_pds = data_request.physical_data_sources[0]
-                connection = Connections.get_by_name(request_pds, True)
-
-                query_runtime_name = Connections.get_connection_engine(connection).dialect.name
-                query_runtime = EnumHelper.from_value_safe(HuskyQueryRuntime, query_runtime_name)
-                if query_runtime is None:
-                    raise UnsupportedSQLOutputException(query_runtime_name)
-
-                return cls(query_runtime)
-            elif len(data_request.physical_data_sources) > 1:
-                raise TooManyPhysicalDataSourcesException(data_request.physical_data_sources)
-        else:
-            return cls(HuskyQueryRuntime.snowflake)
+    def from_request(
+        cls,
+        data_request: Union[BlendingDataRequest, InternalDataRequest],
+        default_runtime: Optional[HuskyQueryRuntime] = None,
+    ):
+        if default_runtime:
```

Comment:
this is rather strange behaviour. Personally, if I define `default_value`, then I expect the code to:

- first try to determine the value "the usual way"
- then, if it wasnt possible, I use the default value

This feels more like `force_runtime`. It is a slight change in name, but it is much easier to understand (from my perspective). Thoughts?